src/api/walker.js

```diff
@@ -84,15 +84,21 @@ Walker.prototype.processDirents = function processDirents(
     // perf: we can avoid entering the condition block if .withSymlinks is not set
     // by using symlinkResolver !== fns.empty; this helps us avoid wasted allocations -
     // which are probably very minor but still.
-    else if (dirent.isSymbolicLink() && this.symlinkResolver !== fns.empty) {
-      let path = fns.joinPathWithBasePath(dirent.name, directoryPath);
-      this.symlinkResolver(path, this.state, (stat, resolvedPath) => {
-        if (stat.isFile()) {
-          this.pushFile(this, resolvedPath, files);
-        } else if (stat.isDirectory()) {
-          this.walkDir(this, resolvedPath, dirent.name, currentDepth - 1);
-        }
-      });
+    else if (dirent.isSymbolicLink()) {
```

Comment:
Instead of having nested conditions, we can move the symlink check to the first place. So:

1. if dirent isSymbolicLink
2. else if dirent isDirectory
3. else treat everything as a file

This will help us avoid duplicate code.